### PR TITLE
grt: Algorithm should not change with execution time

### PR DESCRIPTION
**Pull request by antonblanchard**
Remove some strange code which measures the time taken to execute a
chunk of code, and then makes decisions based on that.

Signed-off-by: Anton Blanchard <anton@linux.ibm.com>

### CODE REVIEW

**Comment by antonblanchard** [comment]
A lot of the CI fails look like small differences in grt stats output, which is expected.

**Comment by eder-matheus** [comment]
@antonblanchard I'll start a secure ci run with your updates before approving this PR. Meanwhile, could you update the ok files for the grt failing tests?

**Comment by maliberty** [comment]
The changes look fine to me - I'll leave it to you and Eder to work out the CI issues and approve.

**Comment by antonblanchard** [comment]
@eder-matheus Could do with some double checking. I had to manually do `save_guideok` for each test that mentioned differences in the test output, rerun the regression tests and finally `save_ok failures`. Makes me wonder if I am missing something.

File: src/grt/test/silence.ok
Comment (eder-matheus) [inline src/grt/test/silence.ok:12]: You missed only this `guideok`. Update this and the grt tests will be all correctly updated.

**Review by eder-matheus — CHANGES_REQUESTED** [review]

**Comment by eder-matheus** [comment]
Thanks, @antonblanchard. The public CI is green, now I'm just waiting for the secure CI to end.

**Review by eder-matheus — APPROVED** [review]

**Comment by eder-matheus** [comment]
@antonblanchard please update your branch with the latest master branch. We have two new grt tests that require updates in the ok files after your changes.

**Comment by antonblanchard** [comment]
@eder-matheus updated to master

**Pull request merged**
